lib/dialects/mssql/query.js

```diff
@@ -315,6 +315,16 @@ class Query extends AbstractQuery {
       id = id || (results && results[0][autoIncrementField]);
       id = id || (autoIncrementFieldAlias && results && results[0][autoIncrementFieldAlias]);
 
+      const dataValues = this.instance.dataValues;
+      for (let key in dataValues) {
+        if (dataValues.hasOwnProperty(key) && dataValues[key]) {
+          if (dataValues[key].constructor.name === 'literal') {
```

Comment:
I don't think this needs to be specific to `literal`, this could also apply to `fn` etc (anything that's generated serverside) 

I think we can just always map the result back into the instance.